app/domain/validation.py

```diff
@@ -19,6 +19,44 @@
 MIN_LAST_ACTIVITY_LIFETIME_FOR_ADMIN_FORCE_VALIDATION = timedelta(hours=24)
 
 
+# In case an admin wants to validate a mission for an employee who did not yet validate its mission
+# He should not be able to do it, except for two special cases
+def pre_check_validate_mission_by_admin(mission, admin_submitter, for_user):
+    activities_to_validate = mission.activities_for(for_user)
+
+    # Checks if employee has already validated its mission
+    if len(mission.validations_for(for_user)) > 0:
+        return
+
+    # Evacuates case where admin is validating for himself
+    if (
+        for_user.id == admin_submitter.id
+        or mission.submitter_id == admin_submitter.id
+    ):
+        return
+
+    # Special case #1 - mission started more than 10d ago
+    mission_start = activities_to_validate[0].start_time
+    mission_old_enough = (
+        datetime.now() - mission_start
+        > MIN_MISSION_LIFETIME_FOR_ADMIN_FORCE_VALIDATION
+    )
+    if mission_old_enough:
+        return
+
+    # Special case #2 - last activity started more than 24h ago and still running
+    last_activity_start = activities_to_validate[-1].start_time
+    last_activity_is_running = not activities_to_validate[-1].end_time
+    last_activity_long_enough = (
+        datetime.now() - last_activity_start
+        > MIN_LAST_ACTIVITY_LIFETIME_FOR_ADMIN_FORCE_VALIDATION
+    )
+    if last_activity_long_enough and last_activity_is_running:
+        return
+
+    raise MissionNotAlreadyValidatedByUserError()
+
+
 def validate_mission(mission, submitter, for_user, creation_time=None):
     validation_time = datetime.now()
     is_admin_validation = company_admin(submitter, mission.company_id)
```

Comment:
:green_book: We could extract this as an argument. Not sure it's better in all situation though